cdap-app-fabric/src/main/java/co/cask/cdap/operations/OperationalStatsReader.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.operations;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.management.ManagementFactory;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+import javax.management.MBeanAttributeInfo;
+import javax.management.MBeanInfo;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+/**
+ * Reads operational stats collected using {@link OperationalStatsReader}.
+ */
+public class OperationalStatsReader {
+
+  private static final Logger LOG = LoggerFactory.getLogger(OperationalStatsReader.class);
+
+  /**
+   * Reads operational stats with the specified property key and value, grouped by the specified groupByKey.
+   *
+   * @param propertyKey the key that must be contained with the specified value in the stat to be returned
+   * @param propertyValue the value that the specified key must have in the stat to be returned
+   * @param groupByKey an additional key in the stat's property to group the stats by
+   * @return a {@link Map} of the group to a {@link Map} of stats of that group
+   */
+  public Map<String, Map<String, Object>> getStats(String propertyKey, String propertyValue,
+                                                   String groupByKey) throws Exception {
```

Comment:
this method throws a whole bunch of exceptions thrown by the JMX classes. I'm not sure it will be helpful to users of this class. Maybe I can describe *when* exceptions will be thrown.